lib/class-wp-rest-menu-items-controller.php

```diff
@@ -755,16 +760,20 @@ protected function prepare_links( $menu_item ) {
 		}
 
 		$path = '';
+		$type = '';
+		$key  = $menu_item->type;
 		if ( 'post_type' === $menu_item->type ) {
 			$path = rest_get_route_for_post( $menu_item->object_id );
+			$type = get_post_type( $menu_item->object_id );
 		} elseif ( 'taxonomy' === $menu_item->type ) {
 			$path = rest_get_route_for_term( $menu_item->object_id );
+			$type = get_term_field( 'taxonomy', $menu_item->object_id );
 		}
 
-		if ( $path ) {
+		if ( $path && $type ) {
 			$links['https://api.w.org/object'][] = array(
```

Comment:
@TimothyBJacobs did you want to change this?

Reply:
Yeah, let's change the type to `menu-item-object`

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/36177/commits/dc23cdc7dfae8e8c102ef58a8c24444513fa7bf8